Project Euler Problem 1

pstaeuble9 opened the pull request.

pstaeuble9 committed:
Project Euler Problem 1

robkinyon commented on Euler1PS.js:
```diff
@@ -0,0 +1,13 @@
+let sum = 0;
```
This should be scoped to the function. What happens if I want to `findTheSum()` of two max numbers, say 500 and 800?

pstaeuble9 replied:
Starting from above:

- Added a link to the problem in a comment
- By tests, I'm assuming you mean, picking criteria, and then running it through the program to make sure it produces the desired result?
- I added a prompt to allow user input
- I'm not sure how I would do a series of numbers. I image I would need to create an array that could hold the multiple inputs and then loop through the array.
   - I believe this would also go for finding two max numbers as well.

robkinyon commented on Euler1PS.js:
```diff
@@ -0,0 +1,13 @@
+
+function findTheSum(max) {
+    for (let i = 3; i < max; i++) {
+        if (i % 3 === 0 || i % 5 === 0) {
```
Do you want object equality (`===`) or numeric equality (`==`)? It turns out to be the same thing in most Javascript interpreters, but that's by accident. Also, Javascript is one of the few languages where that difference matters.

pstaeuble9 replied:
So there was no type coercion I did object equality, although with this file being so small I don't believe it would have mattered

robkinyon reviewed (commented):
A few thoughts:
* In a comment in the source code, put a link to the problem statement.
* How would you write tests for this function?
* How would you modify this to let me tell you which numbers to select based on modulo?
* What if I wanted to give you a series of numbers and you have to handle 1, 2, 3, or more modulo choices?
* How would you handle bad input, such as `"a"` instead of `1000`?
    * What about `"1000"` instead of `1000`?

pstaeuble9 committed:
Add files via upload

Added in verification of input and the ability to input any number.

I'm not sure how I would write tests for this other than to take numbers I can work and make sure they product the desired result.

In order to go through two max numbers, I imagine I would need to get user input in to an array, and then loop through the array.

robkinyon commented on Euler1PS_v2.js:
```diff
@@ -0,0 +1,25 @@
+
+
+function findTheSum(){
+    let input = prompt("Input a number:");
```
You said in your commit message:

```
I'm not sure how I would write tests for this other than to take numbers I can work and make sure they product the desired result.
```

That's *exactly* what you should do. For example, you know that if the number is 1, then the sum should be 0. So, you might do something like where `find_the_sum()` is your function (untested, but should be close to working):

```javascript
let tests = [
    { "input": 1, "expected": 0 },
    { "input": 3, "expected": 3 },
    { "input": 5, "expected": 8 },
    { "input": 6, "expected": 14 },
];

tests.forEach(testcase => {
    let output = find_the_sum(testcase.input);
    if (output == testcase.output) {
        console.log("Input: " + testcase.input + " - Ok");
    }
    else {
        console.log("Input: " + testcase.input + " - NOT Ok");
    }
});
```
